custom_components/petlibro/devices/feeders/one_rfid_smart_feeder.py

```diff
@@ -0,0 +1,88 @@
+from typing import cast
+
+from .feeder import Feeder
+
+from logging import getLogger
+
+# Configure the logger
+_LOGGER = getLogger(__name__)
+
+class OneRFIDSmartFeeder(Feeder):   
+    async def refresh(self):
+        await super().refresh()
+
+        # Fetch data from API
+        grain_status = await self.api.device_grain_status(self.serial)
+        real_info = await self.api.device_real_info(self.serial)
+
+        # Update internal data
+        self.update_data({
+            "grainStatus": grain_status,
+            "realInfo": real_info
+        })
+
+    @property
+    def remaining_desiccant(self) -> str:
+        return cast(str, self._data.get("remainingDesiccantDays"))
+
+    @property
+    def today_feeding_quantity(self) -> int:
+        quantity = self._data.get("grainStatus", {}).get("todayFeedingQuantity")
+        if not quantity:
+            return 0
+
+        return self.convert_unit(quantity)
+
+    @property
+    def today_feeding_times(self) -> int:
+        return cast(int, self._data.get("grainStatus", {}).get("todayFeedingTimes"))
+
```

Comment:
Could be removed if this inherits from `GranaryFeeder`

```suggestion
```

Reply:
Could yes, but lets review whether these defaults should then be handled in the main feeder instead.